core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabaseAdmin.java

```diff
@@ -1,73 +1,187 @@
+                  RdbEngine.ORACLE,
+                  ImmutableMap.<DataType, String>builder()
+                      .put(DataType.TEXT, "VARCHAR2(64)")
+                      .put(DataType.BLOB, "RAW(64)")
+                      .build())
+              .put(RdbEngine.SQL_SERVER, ImmutableMap.<DataType, String>builder().build())
+              .build();
   private final BasicDataSource dataSource;
-  private final Optional<String> namespacePrefix;
+  private final Optional<String> schemaPrefix;
   private final JdbcTableMetadataManager metadataManager;
+  private final RdbEngine rdbEngine;
 
   @Inject
   public JdbcDatabaseAdmin(JdbcConfig config) {
     dataSource = JdbcUtils.initDataSource(config);
-    namespacePrefix = config.getNamespacePrefix();
-    RdbEngine rdbEngine = JdbcUtils.getRdbEngine(config.getContactPoints().get(0));
-    metadataManager = new JdbcTableMetadataManager(dataSource, namespacePrefix, rdbEngine);
+    schemaPrefix = config.getNamespacePrefix();
+    rdbEngine = JdbcUtils.getRdbEngine(config.getContactPoints().get(0));
+    metadataManager = new JdbcTableMetadataManager(dataSource, schemaPrefix, rdbEngine);
   }
 
   @VisibleForTesting
   public JdbcDatabaseAdmin(
-      JdbcTableMetadataManager metadataManager, Optional<String> namespacePrefix) {
-    dataSource = null;
+      BasicDataSource dataSource,
+      JdbcTableMetadataManager metadataManager,
+      Optional<String> namespacePrefix,
+      RdbEngine rdbEngine) {
+    this.dataSource = dataSource;
     this.metadataManager = metadataManager;
-    this.namespacePrefix = namespacePrefix.map(n -> n + "_");
+    this.schemaPrefix = namespacePrefix;
+    this.rdbEngine = rdbEngine;
   }
 
   @Override
   public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    try (Connection connection = dataSource.getConnection()) {
+      connection.setAutoCommit(false);
+      connection.setTransactionIsolation(Connection.TRANSACTION_SERIALIZABLE);
+      try {
+        createSchemaIfNotExists(connection, namespace);
+        createTableInternal(connection, namespace, table, metadata);
+        createIndex(connection, namespace, table, metadata);
+      } catch (SQLException e) {
+        connection.rollback();
+        throw e;
+      }
+      connection.commit();
+    } catch (SQLException e) {
+      throw new ExecutionException("creating the table failed", e);
+    }
+    metadataManager.addTableMetadata(namespace, table, metadata);
   }
 
   @Override
   public void dropTable(String namespace, String table) throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    dropTableInternal(namespace, table);
+    metadataManager.deleteTableMetadata(namespace, table);
+    if (metadataManager.getTableNames(namespace).isEmpty()) {
+      dropSchema(namespace);
+    }
   }
```

Comment:
I added the new method `TableMetadataManager.getTablesNames()` because I needed a way to know if there was any tables associated with the given namespace. I figured this was more proper to rely on the `MetadataManager` data than run specific sql queries to check if a namespace contains some tables.